File: bubble_sort.py
def bubble_sort(itr, done=False):
  '''Takes an iterable and returns it sorted'''
  while not done:
    *itr, done = bblsrt(itr)
  return itr

def bblsrt(itr, done=True):
  '''Recursive logic'''
  if(len(itr) == 1): return itr[0], done
  a, b, *rest = itr
  if(done): done = a <= b
  return min(a,b), *bblsrt((max(a,b), *rest), done)

File: bubble_sort.js
// BUBBLE SORT JS

function bubbleSort(arr, sorted=false) {
    //Takes an array and returns it sorted
    while(!sorted){
      arr = bblSrt(arr);
      sorted = arr.pop()
    }
    return arr
}

function bblSrt(arr, sorted=true, r=[]){
  // Recursive logic
  if(arr.length === 1) return [arr.pop(), sorted]
  let [ith, next, ...rest] = arr;
  min = ith <= next ? ith : next;
  max = ith > next ? ith : next;
  sorted = sorted ? ith <= next : sorted;
  return r.concat([min, ...bblSrt([max, ...rest], sorted)])
} 

